test/cpp/end2end/client_fork_test.cc

```diff
@@ -0,0 +1,176 @@
+// Copyright 2022 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <grpc/support/port_platform.h>
+
+#ifndef GRPC_ENABLE_FORK_SUPPORT
+// No-op for builds without fork support.
+int main(int /* argc */, char** /* argv */) { return 0; }
+#else  // GRPC_ENABLE_FORK_SUPPORT
+
+#include <signal.h>
+
+#include <gtest/gtest.h>
+
+#include "absl/strings/str_cat.h"
+
+#include <grpc/fork.h>
+#include <grpc/grpc.h>
+#include <grpc/support/log.h>
+#include <grpc/support/time.h>
+#include <grpcpp/channel.h>
+#include <grpcpp/client_context.h>
+#include <grpcpp/create_channel.h>
+#include <grpcpp/server.h>
+#include <grpcpp/server_builder.h>
+#include <grpcpp/server_context.h>
+
+#include "src/core/lib/gprpp/fork.h"
+#include "src/proto/grpc/testing/echo.grpc.pb.h"
+#include "test/core/util/port.h"
+#include "test/core/util/test_config.h"
+#include "test/cpp/util/test_config.h"
+
+namespace grpc {
+namespace testing {
+namespace {
+
+using ::grpc::testing::EchoRequest;
```

Comment:
This is already in the grpc::testing namespace. Is there ambiguity here?

Reply:
Good catch. Removed.